docs/upgrading_to_v21.0.md

```diff
@@ -0,0 +1,16 @@
+# Upgrading to v21.0
+
+The v21.0 release of *kubernetes-engine* is a backwards incompatible
+release for the Anthos Service Mesh (ASM) module.
+
+### Terraform Kubernetes Engine Module
+
+The [Terraform Kubernetes Engine Module](https://github.com/terraform-google-modules/terraform-google-kubernetes-engine) has been rewritten to use the 'kubernetes_config_map_v1_data' resouce added to the Terraform Kubernetes provider version 2.10.
+
+1. Run `terraform state rm module.gke.kubernetes_config_map.kube-dns`
+2. Update the module version to v21.0
```

Comment:
Are any diffs expected?

Reply:
I don't believe so and ran a quick test from v20 to v21 using the above procedure.  [Kubernetes_config_map_v1_data](https://registry.terraform.io/providers/hashicorp/kubernetes/latest/docs/resources/config_map_v1_data) updates data in a pre-existing ConfigMap, so regardless if the ConfigMap was created by gke/addon-manager or the v20 module's config_map it shouldn't matter.  With the `force=true` it will simply (re-)update the ConfigMap data.  Even if the state isn't dropped, worst case the ConfigMap data will be quickly updated to the declared configuration.  Let me know if any questions or concerns.

Reply:
SGTM, thanks for investigating!